src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -866,15 +866,29 @@ private void recalculateIsland(@NotNull Island island, @NotNull List<Chunk> chun
                 }
             }
         });
-        chunks.forEach(chunk -> {
-            for (BlockState blockState : chunk.getTileEntities()) {
-                if (!(blockState instanceof CreatureSpawner)) continue;
-                if (!island.isInIsland(blockState.getLocation())) continue;
-                CreatureSpawner creatureSpawner = (CreatureSpawner) blockState;
-                IslandSpawners islandSpawners = IridiumSkyblock.getInstance().getIslandManager().getIslandSpawners(island, creatureSpawner.getSpawnedType());
-                islandSpawners.setAmount(islandSpawners.getAmount() + 1);
-            }
-        });
+        if (Bukkit.isPrimaryThread() == false) {
```

Comment:
Change this to
```java
if(Bukkit.isPrimaryThread()){

}else{
Bukkit.getScheduler().runTask(IridiumSkyblock.getInstance(), this::recalculateIsland);
}
```
since its kinda dum to repeat the same code twice

Reply:
No need to go through everything synchronously because it is only calling the entity and it risks duplicating the data because it would be necessary to re-reset the cache.

Reply:
I'm create a function for it